internal/core/core.go

```diff
@@ -717,6 +720,112 @@ func (cs *coreService) DeleteMessage(ctx context.Context, options DeleteMessageO
 	return nil
 }
 
+// EditMessage edits an existing message in a room.
+func (cs *coreService) EditMessage(ctx context.Context, options EditMessageOptions) error {
+	if options.Text == "" {
+		return errors.New("You must provide some text for the message")
+	}
+
+	if options.SenderID == "" {
+		return errors.New("You must provide the ID of the user editing the message")
+	}
+
+	if options.RoomID == "" {
+		return errors.New("You must provide the ID of the room in which the message to edit belongs")
+	}
+
+	requestBody, err := common.CreateRequestBody(map[string]string{"text": options.Text})
+	if err != nil {
+		return err
+	}
+
+	response, err := common.RequestWithUserToken(
+		cs.underlyingInstance,
+		ctx,
+		options.SenderID,
+		client.RequestOptions{
+			Method: http.MethodPut,
+			Path:   fmt.Sprintf("/rooms/%s/messages/%d", options.RoomID, options.MessageID),
+			Body:   requestBody,
+		})
+	if err != nil {
+		return err
+	}
+	defer response.Body.Close()
+
+	return nil
+}
+
+// EditSimpleMessage edits an existing message in a room, replacing it with a simple message.
+func (cs *coreService) EditSimpleMessage(ctx context.Context, options EditSimpleMessageOptions) error {
+	return cs.EditMultipartMessage(ctx, EditMultipartMessageOptions{
+		RoomID:    options.RoomID,
+		MessageID: options.MessageID,
+		SenderID:  options.SenderID,
+		Parts:     []NewPart{NewInlinePart{Type: "text/plain", Content: options.Text}},
+	})
+}
+
+// EditMultipartMessage edits an existing message in a room, replacing it with a multipart message.
+func (cs *coreService) EditMultipartMessage(ctx context.Context, options EditMultipartMessageOptions) error {
+	if len(options.Parts) == 0 {
+		return errors.New("You must provide at least one message part")
+	}
+
+	if options.SenderID == "" {
+		return errors.New("You must provide the ID of the user editing the message")
+	}
+
+	if options.RoomID == "" {
+		return errors.New("You must provide the ID of the room in which the message to edit belongs")
+	}
+
+	requestParts := make([]interface{}, len(options.Parts))
+	g := errgroup.Group{}
```

Comment:
Good use of errgroup!

Given that have a context, we can use it in the error group `WithContext`:
```suggestion
	g, gCtx := errgroup.WithContext(ctx)
```